internal/service/iam/access_key.go

```diff
@@ -0,0 +1,344 @@
+		AccessKeyId: createResp.AccessKey.AccessKeyId,
+		CreateDate:  createResp.AccessKey.CreateDate,
+		Status:      createResp.AccessKey.Status,
+		UserName:    createResp.AccessKey.UserName,
+	})
+}
+
+func resourceAccessKeyRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).IAMConn
+
+	request := &iam.ListAccessKeysInput{
+		UserName: aws.String(d.Get("user").(string)),
+	}
+
+	getResp, err := conn.ListAccessKeys(request)
+	if err != nil {
+		if iamerr, ok := err.(awserr.Error); ok && iamerr.Code() == "NoSuchEntity" { // XXX TEST ME
+			// the user does not exist, so the key can't exist.
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading IAM access key: %s", err)
+	}
+
+	for _, key := range getResp.AccessKeyMetadata {
+		if key.AccessKeyId != nil && *key.AccessKeyId == d.Id() {
+			return resourceAccessKeyReadResult(d, key)
+		}
+	}
+
+	// Guess the key isn't around anymore.
+	d.SetId("")
+	return nil
+}
+
+func resourceAccessKeyReadResult(d *schema.ResourceData, key *iam.AccessKeyMetadata) error {
+	d.SetId(aws.StringValue(key.AccessKeyId))
+
+	if key.CreateDate != nil {
+		createDate := key.CreateDate
+		expirationDate := createDate.Add(time.Duration(d.Get("max_age").(int)) * time.Second)
+
+		d.Set("create_date", aws.TimeValue(createDate).Format(time.RFC3339))
+		d.Set("expiration_date", expirationDate.Format(time.RFC3339))
+	} else {
+		d.Set("create_date", nil)
+		d.Set("expiration_date", nil)
+	}
+
+	d.Set("status", key.Status)
+	d.Set("user", key.UserName)
+
+	return nil
+}
+
+func resourceAccessKeyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).IAMConn
+
+	if d.HasChange("status") {
+		if err := resourceAccessKeyStatusUpdate(conn, d); err != nil {
+			return err
+		}
+	}
+
+	return resourceAccessKeyRead(d, meta)
+}
+
+func resourceAccessKeyDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).IAMConn
+
+	request := &iam.DeleteAccessKeyInput{
+		AccessKeyId: aws.String(d.Id()),
+		UserName:    aws.String(d.Get("user").(string)),
+	}
+
+	if _, err := conn.DeleteAccessKey(request); err != nil {
+		return fmt.Errorf("Error deleting access key %s: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func resourceAccessKeyStatusUpdate(conn *iam.IAM, d *schema.ResourceData) error {
+	request := &iam.UpdateAccessKeyInput{
+		AccessKeyId: aws.String(d.Id()),
+		Status:      aws.String(d.Get("status").(string)),
+		UserName:    aws.String(d.Get("user").(string)),
+	}
+
+	if _, err := conn.UpdateAccessKey(request); err != nil {
+		return fmt.Errorf("Error updating access key %s: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func hmacSignature(key []byte, value []byte) ([]byte, error) {
+	h := hmac.New(sha256.New, key)
+	if _, err := h.Write(value); err != nil {
+		return []byte(""), err
+	}
+	return h.Sum(nil), nil
+}
+
+func SessmTPPasswordFromSecretKeySigV4(key *string, region string) (string, error) {
```

Comment:
Should this be `SesSMTP` ?

```suggestion
func SesSMTPPasswordFromSecretKeySigV4(key *string, region string) (string, error) {
```

Reply:
This is actually taken from the official AWS provider, so I'm copying their apparent typo for consistency... :\